packages/workflow/src/WorkflowDataProxy.ts

```diff
@@ -792,7 +792,11 @@ export class WorkflowDataProxy {
 			nodeCause: string,
 			usedMethodName: 'itemMatching' | 'pairedItem' | 'item' | '$getPairedItem' = 'pairedItem',
 		) => {
-			const message = `Using the ${usedMethodName} method doesn't work with pinned data in this scenario. Please unpin '${nodeCause}' and try again.`;
+			const pinData = getPinDataIfManualExecution(that.workflow, nodeCause, that.mode);
+			const message = pinData
+				? `Using the ${usedMethodName} method doesn't work with pinned data in this scenario. Please unpin '${nodeCause}' and try again.`
+				: `Missing ${usedMethodName} data (node '${nodeCause}' probably didn't supply it)`;
```

Comment:
I didn't think the message was very clear. Is this better?

```suggestion
				: `Paired item data for ${usedMethodName} from node '${nodeCause}' is unavailable. Ensure '${nodeCause}' is providing the required output.`;
```